components/ui-api-layer/internal/domain/k8s/status/pod.go

```diff
@@ -0,0 +1,77 @@
+package status
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/kyma/components/ui-api-layer/internal/gqlschema"
+	v1 "k8s.io/api/core/v1"
+)
+
+type PodExtractor struct{}
+
+func (ext *PodExtractor) ContainerStatusesToGQLContainerStates(in []v1.ContainerStatus) []gqlschema.ContainerState {
```

Comment:
Hm... this is state extractor, not status ;) My bad, I suggested moving it to `status`.

So move it to `state` package and rename method to `States` and it will be `ContainerExtractor` :)